src/services/api.service.ts

```diff
@@ -150,6 +133,45 @@ export const getSeriesByMediaIds = async (mediaIds: string[]): Promise<{ [key in
   return await getDataOrThrow(response);
 };
 
+// Max count per request
+const MAX_EPISODES_COUNT = 50;
+
+/**
+ * Get all episodes of the selected series
+ * @param {string} seriesId
+ */
+export const getEpisodes = async (seriesId: string | undefined): Promise<{ [id: string]: PlaylistItem }> => {
+  if (!seriesId) {
+    throw new Error('Series ID is required');
+  }
+
+  const pathname = `/apps/series/${seriesId}/episodes`;
+  const url = addQueryParams(`${import.meta.env.APP_API_BASE_URL}${pathname}`, {});
+
+  const response = await fetch(url);
+  const data = await getDataOrThrow(response);
+
+  let media = data.media;
+
+  // We will send additional requests in case there are more than 50 items
+  if (data.total > MAX_EPISODES_COUNT) {
```

Comment:
I'm not sure this pagination approach is going to work. When series have 1000's of episodes we'll sit here fetching all of them before we can proceed, no?

Ideally we should just grab the first 50 we want to show and then load more in the background or on virtualized scroll or something. It might be nice to ask the BE to add a season filter to the episodes endpoint to make life easier too.

Reply:
Added new endpoints.